account_move_name_sequence/models/ir_sequence.py

```diff
@@ -17,11 +17,11 @@ def _create_date_range_seq(self, date):
             date_from = date_obj
             date_to = date_obj
         elif "%(range_month)s" in prefix_suffix:
-            date_from = date_obj + relativedelta(day=1)
-            date_to = date_obj + relativedelta(day=31)
+            date_from = fields.Date.start_of(date_obj, "month")
+            date_to = fields.Date.end_of(date_obj, "month")
         else:
-            date_from = date_obj + relativedelta(day=1, month=1)
-            date_to = date_obj + relativedelta(day=31, month=12)
+            date_from = fields.Date.start_of(date_obj, "year")
```

Comment:
IMHO this refactoring should be done from v14.0 instead then FW to v15.0

In order to avoid diverging both versions

I mean, v14.0 supports the same method

Or better let it as it in v14.0 without changes in order to avoid this process